docs/design/run.md

```diff
@@ -0,0 +1,278 @@
+enables incremental builds (e.g by letting cargo reuse its `target/` directory).
+However, it also means that runs potentially become less reproducible. We will 
+provide a flag for removing ignored files from the temporary working copies to
+address that. 
+
+Another problem with leaving ignored files in the temporary directories is that
+they take up space. That is especially problematic in the case of cargo (the 
+`target/` directory often takes up tens of GBs). The same flag for cleaning up
+ignored files can be used to address that. We may want to also have a flag for 
+cleaning up temporary working copies *after* running the command. 
+
+An early version of the command will directly use [Treestate] to 
+to manage the temporary working copies. That means that running `jj` inside the 
+temporary working copies will not work . We can later extend that to use a full
+[Workspace]. To prevent operations in the working copies from 
+impacting the repo, we can use a separate [OpHeadsStore] for it.
+
+### Modifying the Working Copy
+
+Since the subprocesses will run in temporary working copies by default, they 
+won't interfere with the user's working copy. The user can therefore continue
+to work in it while `jj run` is running. 
+
+We want subprocesses to be able to make changes to the repo by updating their
+assigned working copy. Let's say the user runs `jj run` on just commits A and 
+B, where B's parent is A. Any changes made on top of A would be squashed into 
+A, forming A'. Similarly B' would be formed by squasing it into B. We can then
+either do a normal rebase of B' onto A', or we can simply update its parent to
+A'. The former is useful, e.g when the subprocess only makes a partial update
+of the tree based on the parent commit. In addition to these two modes, we may 
+want to have an option to ignore any changes made in the subprocess's working 
+copy.
+
+### Modifying the Repo
+
+Once we give the subprocess access to a fork of the repo via separate 
+[OpHeadsStore], it will be able to create new operations in its fork.
+If the user runs `jj run -r foo` and the subprocess checks out another commit,
+it's not clear what that should do. We should probably just verify that the 
+working-copy commit's parents are unchanged after the subprocess returns. Any
+operations created by the subprocess will be ignored. 
+
+### Rewriting the revisions 
+
+Like all commands, `jj run` will refuse to rewrite public/immutable commits.
+
+### Public revisions
+
+For published revisions, we will not allow `jj run`  to modify them and then 
+immediately error out, as published history should be immutable. We may want to
+support a `--force` flag for an override but it won't be available in the first
+iteration of the command. 
+
+### Private/Draft revisions
+
+For private/draft revisions, we just amend the changes, as Jujutsu usually does. 
+We also expose the actual behavior as a command option.
```

Comment:
I think you can delete this now. It's covered under "Rewriting the revisions" AFAICT. I don't see anything specific to `jj run` here.

Reply:
Dropped the section, while keeping some part of the last sentence in `private/draft revisions` in a second sentence.